lib/services/metadata_provider.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:finamp/models/finamp_models.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:get_it/get_it.dart';
+import 'package:logging/logging.dart';
+
+import '../models/jellyfin_models.dart';
+import 'downloads_service.dart';
+import 'finamp_settings_helper.dart';
+import 'jellyfin_api_helper.dart';
+
+final metadataProviderLogger = Logger("MetadataProvider");
+
+class MetadataRequest {
+  const MetadataRequest({
+    required this.item,
+    this.includeLyrics = false,
+  }) : super();
+
+  final BaseItemDto item;
+
+  final bool includeLyrics;
+
+  @override
+  bool operator ==(Object other) {
+    return other is MetadataRequest &&
+        other.includeLyrics == includeLyrics &&
+        other.item.id == item.id;
+  }
+
+  @override
+  int get hashCode => Object.hash(item.id, includeLyrics);
+}
+
+enum MetadataState {
+  loading,
```

Comment:
You shouldn't need this.  Riverpod has this built in.  Watching the provider gives you an AsyncValue, which has an isLoading attribute.  Also, is it actually possible to return a MetadataProvider in a loading state in this setup?